### PR TITLE
Sign up service

### PR DESCRIPTION
**Pull request by miguelhenaoa**
## Summary

- Add API configuration
- Create UsersServive
- Call API from view
## Screenshots

[Change!] Show the screenshots of the views you modified.
image

## Trello Card

https://trello.com/c/OEPmfU90

### CODE REVIEW

File: src/config/api.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { create } from 'apisauce';
+import { CamelcaseSerializer, SnakecaseSerializer } from 'cerealizr';
 
-const baseURL = 'http://wolox.com';
+const baseURL = 'https://books-training-rails.herokuapp.com/api/v1' || 'http://wolox.com';
```
Comment (TaloMontes) [inline src/config/api.js:4]: delete wolox url

**Review by TaloMontes — DISMISSED** [review]

File: src/components/App/index.tsx [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import React from 'react';
+import { QueryClientProvider, QueryClient } from 'react-query';
 
 import 'scss/application.scss';
 import Home from 'screens/Home/index';
```
Comment (alejobh) [inline src/components/App/index.tsx:5]: remove index

File: src/config/api.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import { create } from 'apisauce';
+import { CamelcaseSerializer, SnakecaseSerializer } from 'cerealizr';
 
-const baseURL = 'http://wolox.com';
+const baseURL = 'https://books-training-rails.herokuapp.com/api/v1';
```
Comment (alejobh) [inline src/config/api.js:4]: this should be placed on a .env variable normally called `REACT_APP_API_BASE_URL`

File: src/screens/SignUp/index.tsx [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 import { useForm } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
+import { useMutation } from 'react-query';
 
 import FormCard from '../../components/FormCard';
 import Input from '../../components/Input';
 import { User } from '../../utils/types';
 import { regexEmail, regexPassword } from '../../constants/forms';
+import { signUp } from '../../services/UsersService';
+import Loading from '../../components/Spinner/components/loading';
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:11]: use absolute routes

File: src/screens/SignUp/index.tsx [hunk 5]
```diff
@@ -14,10 +17,20 @@ function SignUp() {
...
   const { t } = useTranslation();
+  const [errorMsg, setErrorMsg] = useState('');
+  const { isLoading, isError, mutate } = useMutation((data: User) => signUp(data), {
+    onSuccess: res => {
+      // eslint-disable-next-line no-console
+      console.log('res', res);
+    },
+    onError: () => {
+      setErrorMsg(t('ErrorRequest:network'));
+    }
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:29]: this could be different, use problem or code from the request error

File: src/screens/SignUp/index.tsx [hunk 6]
```diff
@@ -14,10 +17,20 @@ function SignUp() {
...
+  });
+  const onSubmit = () => {
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:31]: onSubmit gives you the values as function param

File: src/screens/SignUp/index.tsx [hunk 7]
```diff
@@ -14,10 +17,20 @@ function SignUp() {
     getValues,
     formState: { errors }
   } = useForm<User>();
-  // eslint-disable-next-line no-console
-  const onSubmit = (data: User) => console.log(data);
-
   const { t } = useTranslation();
+  const [errorMsg, setErrorMsg] = useState('');
+  const { isLoading, isError, mutate } = useMutation((data: User) => signUp(data), {
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:22]: `const { isLoading, isError, mutate } = useMutation(signUp, {`

File: src/screens/SignUp/index.tsx (hunk 6 above)
Comment (alejobh) [reply src/screens/SignUp/index.tsx:31]: `const onSubmit = (values: User) => {`

File: src/screens/SignUp/index.tsx [hunk 8]
```diff
@@ -72,6 +85,14 @@ function SignUp() {
             error={errors.passwordConfirmation?.message}
           />
 
+          {isError && <span className="form-alert">{errorMsg}</span>}
+
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:89]: avoid blank lines on the jsx

File: src/screens/SignUp/index.tsx [hunk 9]
```diff
@@ -72,6 +85,14 @@ function SignUp() {
...
+          {isLoading && (
+            <div className="row full-width center">
+              <Loading name="circle" />
+            </div>
+          )}
+
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:95]: same

File: src/screens/SignUp/index.tsx [hunk 10]
```diff
@@ -72,6 +85,14 @@ function SignUp() {
             error={errors.passwordConfirmation?.message}
           />
```
Comment (alejobh) [inline src/screens/SignUp/index.tsx:87]: and this

**Review by alejobh — CHANGES_REQUESTED** [review]
nice work!

**Review by alejobh — APPROVED** [review]

**Review by alejobh — APPROVED** [review]

**Pull request merged**